src/cascadia/TerminalApp/ActionArgs.cpp

```diff
@@ -15,3 +15,186 @@
+    }
+
+    winrt::hstring CopyTextArgs::GenerateName()
+    {
+        if (_SingleLine)
+        {
+            return RS_(L"CopyTextAsSingleLineCommandKey");
+        }
+        return RS_(L"CopyTextCommandKey");
+    }
+
+    winrt::hstring NewTabArgs::GenerateName()
+    {
+        winrt::hstring newTerminalArgsStr;
+        if (_TerminalArgs)
+        {
+            newTerminalArgsStr = _TerminalArgs.GenerateName();
+        }
+        if (newTerminalArgsStr.empty())
+        {
+            return RS_(L"NewTabCommandKey");
+        }
+        else
+        {
+            return winrt::hstring{ fmt::format(L"{}, {}", RS_(L"NewTabCommandKey"), newTerminalArgsStr) };
+        }
+        return L"NewTabArgs";
+    }
+
+    winrt::hstring SwitchToTabArgs::GenerateName()
+    {
+        return winrt::hstring{ fmt::format(L"{}, index:{}", RS_(L"SwitchToTabCommandKey"), _TabIndex) };
+    }
+
+    winrt::hstring ResizePaneArgs::GenerateName()
+    {
+        winrt::hstring directionString;
+        switch (_Direction)
+        {
+        case Direction::Left:
+            directionString = RS_(L"DirectionLeft");
+            break;
+        case Direction::Right:
+            directionString = RS_(L"DirectionRight");
+            break;
+        case Direction::Up:
+            directionString = RS_(L"DirectionUp");
+            break;
+        case Direction::Down:
+            directionString = RS_(L"DirectionDown");
+            break;
+        }
+        return winrt::hstring{ fmt::format(RS_(L"ResizePaneWithArgCommandKey").c_str(), directionString) };
+    }
+
+    winrt::hstring MoveFocusArgs::GenerateName()
+    {
+        winrt::hstring directionString;
+        switch (_Direction)
+        {
+        case Direction::Left:
+            directionString = RS_(L"DirectionLeft");
+            break;
+        case Direction::Right:
+            directionString = RS_(L"DirectionRight");
+            break;
+        case Direction::Up:
+            directionString = RS_(L"DirectionUp");
+            break;
+        case Direction::Down:
+            directionString = RS_(L"DirectionDown");
+            break;
+        }
+        return winrt::hstring{ fmt::format(RS_(L"MoveFocusWithArgCommandKey").c_str(), directionString) };
+    }
+
+    winrt::hstring AdjustFontSizeArgs::GenerateName()
+    {
+        if (_Delta < 0)
+        {
+            return _Delta == -1 ? winrt::hstring{ fmt::format(RS_(L"DecreaseFontSizeCommandKey").c_str()) } :
+                                  winrt::hstring{ fmt::format(RS_(L"DecreaseFontSizeWithAmountCommandKey").c_str(), -_Delta) };
+        }
+        else
+        {
+            return _Delta == 1 ? winrt::hstring{ fmt::format(RS_(L"IncreaseFontSizeCommandKey").c_str()) } :
+                                 winrt::hstring{ fmt::format(RS_(L"IncreaseFontSizeWithAmountCommandKey").c_str(), _Delta) };
+        }
+    }
+
+    winrt::hstring SplitPaneArgs::GenerateName()
+    {
+        std::wstringstream ss;

```

Comment:
Do you cache these somewhere? I understand you have to generate them to work with the permutations of these types of actions, but I can't seem to find a cache of the resultant strings. I'm concerned about regenerating them on every open of the command palette. Am I missing something?

Reply:
Oh yea, I can clear that up. These names are only generated for each action when we parse the settings. So they're generated once per action, per settings reload. Once the strings are generated, we'll hang on to them in the Command object (in it's `Name`) until the settings are reloaded.